esp/public/media/scripts/onsite/ajax_status.js

```diff
@@ -201,9 +248,10 @@ function print_schedule()
     result = $j.ajax({
         url: printing_url,
         data: student_data,
-        async: false
+        success: function (data, text_status, jqxhr) {
+            add_message(data.message);
```

Comment:
I think that when I was testing this (and on the site too, evidently), jQuery automagically parses `data` as JSON based on the mime type.

Glancing at the docs, it looks like I can specify `dataType: 'json'` explicitly, so I'll do that (although I bet there are lots of other places that use `$.ajax` and don't).